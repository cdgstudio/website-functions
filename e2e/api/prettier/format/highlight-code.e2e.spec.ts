import { describe, expect, it } from '@jest/globals';

describe('/api/code/highlight', () => {
  it('should highlight the code', async () => {
    const query = new URLSearchParams({
      language: 'typescript',
    });

    const response = await fetch(new URL(`http://localhost:8888/api/code/highlight?${query}`), {
      method: 'POST',
      body: `class A {
        constructor() {}
      }`,
    });

    const text = await response.text();

    expect(text).toMatchSnapshot();
  });
});
