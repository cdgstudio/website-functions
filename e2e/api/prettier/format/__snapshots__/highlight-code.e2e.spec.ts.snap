// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/code/highlight should highlight the code 1`] = `
"<span class="hljs-keyword">class</span> <span class="hljs-title class_">A</span> {
        <span class="hljs-title function_">constructor</span>(<span class="hljs-params"></span>) {}
      }"
`;
